public/app/features/live/pipeline/models.gen.ts

```diff
@@ -0,0 +1,116 @@
+/* Do not change, this code is generated from Golang structs */
+
+import { FieldConfig } from '@grafana/data';
+
+export interface JsonFrameConverterConfig {}
+export interface AutoInfluxConverterConfig {
+  frameFormat: string;
+}
+export interface ExactJsonConverterConfig {
+  fields: Field[];
+}
+export interface Label {
+  name: string;
+  value: string;
+}
+export interface Field {
+  name: string;
+  type: number;
+  value: string;
+  labels?: Label[];
+  config?: FieldConfig;
+}
+export interface AutoJsonConverterConfig {
+  fieldTips?: { [key: string]: Field };
+}
+export interface ConverterConfig {
+  type: Omit<keyof ConverterConfig, 'type'>;
+  jsonAuto?: AutoJsonConverterConfig;
+  jsonExact?: ExactJsonConverterConfig;
+  influxAuto?: AutoInfluxConverterConfig;
+  jsonFrame?: JsonFrameConverterConfig;
+}
+export interface MultipleFrameProcessorConfig {
+  processors: FrameProcessorConfig[];
+}
+export interface KeepFieldsFrameProcessorConfig {
+  fieldNames: string[];
+}
+export interface DropFieldsFrameProcessorConfig {
+  fieldNames: string[];
+}
+export interface FrameProcessorConfig {
+  type: Omit<keyof FrameProcessorConfig, 'type'>;
```

Comment:
wait, that's `Omit<keyof FrameProcessorConfig, 'type'>`, that is `dropFields | keepFields | multiple`? shouldn't it be just a `string` or an enum?

Reply:
ohhh - I see, that's correct, it acts like a "one of" - we will be able to do `frameProcessorConfig[frameProcessorConfig.type]`